### PR TITLE
Allow flutter engine path to be set when using run_mojo

### PR DESCRIPTION
**Pull request by krisgiesing**
Useful when doing development that requires changes to both mojo and flutter engine code.

### CODE REVIEW

**Comment by eseidelGoogle** [comment]
Who owns this code now that @iansf is gone?  I'm lacking context here, but this does seem useful.

**Comment by krisgiesing** [comment]
I talked to @jamesr and @abarth about this last week, not sure who owns the code long-term.

**Comment by abarth** [comment]
@eseidelGoogle I'm the owner for the flutter_tools package generally, but the run_mojo command is still somewhat of a mystery to me.  I'm slowly learning how it does its magic.

File: packages/flutter_tools/lib/src/commands/run_mojo.dart [hunk 1]
```diff
@@ -58,6 +59,18 @@ class RunMojoCommand extends Command {
...
+    BuildConfiguration result;
+    TargetPlatform target = getCurrentHostPlatformAsTarget();
+    for (BuildConfiguration config in buildConfigurations) {
+      if (config.targetPlatform == target) {
+        result = config;
+        break;
+      }
+    }
+    return result;
+  }
```
Comment (abarth) [inline packages/flutter_tools/lib/src/commands/run_mojo.dart]: This code is duplicated from somewhere, right?  Maybe refactor so it can be shared?

**Comment by krisgiesing** [comment]
Travis seems to have gotten lost running pub upgrade, which I don't think would be related to this change. Is there some way to re-trigger the tests?

**Comment by abarth** [comment]
LGTM

File: packages/flutter_tools/lib/src/commands/run_mojo.dart (hunk 1 above)
Comment (krisgiesing) [inline packages/flutter_tools/lib/src/commands/run_mojo.dart]: It's not duplicated as far as I know. I could put it somewhere if it seems useful.

Comment (abarth) [inline packages/flutter_tools/lib/src/commands/run_mojo.dart]: If its not duplicated, then lets leave it here and move it once there's a second client.

**Comment by krisgiesing** [comment]
Needs revision, latest test failure is valid.

**Pull request merged**